pages/_app.js

```diff
@@ -51,6 +51,32 @@ const connectUser = async () => {
   
 }
 
+const handleMagicLinkWeb3 = async (provider) => {
+  try{
+    const {idx, ceramic, seed} = await generateIDXForMagic(provider);
```

Comment:
Can we reuse the previous function instead of creating the different one as the provider is the only difference here?

Reply:
Sure, will do that